ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js

```diff
@@ -44,14 +46,23 @@ const AdvancedGasFeeDefaults = () => {
     if (isDefaultSettingsSelected) {
       dispatch(setAdvancedGasFee(null));
       setDefaultSettingsSelected(false);
+      updateTransactionEventFragment({
+        advanced_gas_defaults_updated: true,
+        advanced_gas_defaults_updated_maxbasefee: null,
+        advanced_gas_defaults_updated_priorityfee: null,
+      });
     } else {
       dispatch(
         setAdvancedGasFee({
           maxBaseFee,
           priorityFee: maxPriorityFeePerGas,
         }),
       );
-      setDefaultSettingsSelected(true);
+      updateTransactionEventFragment({
+        advanced_gas_defaults_updated: true,
+        advanced_gas_defaults_updated_maxbasefee: maxBaseFee,
+        advanced_gas_defaults_updated_priorityfee: maxPriorityFeePerGas,
+      });
```

Comment:
Above takes care to add/remove parameter in transaction fragment for advance gas default updates. This is exactly not part of transaction flow but more of user settings. But I still added it to transaction, I also see it included in transaction metrics [here](https://www.notion.so/EIP-1559-v2-330b809eb95048f9a45fa70c9f9b9303#3dea90b44d7e444ebd74190a2095fa75).

Reply:
this is fine for now, but as its a user level setting (or is it per account?) we'll want to migrate these to traits whenever we do the work to make that a thing. Sory for a future date. 

I do think gas_default_max_base_fee is sufficient, the presence of it in the payload is good enough to track that it was updated.

Reply:
These is user level and updated preference controller.